File: src/js/09_listy_wprowadzenie/zad10.js
/*
ZAD-10 — Czy punkty mogą być wierzchołkami trójkąta?

**Poziom:** ★★☆
**Tagi:** `geometria`, `warunki`, `listy`

### Treść

Wczytaj współrzędne trzech punktów `A(xA, yA)`, `B(xB, yB)`, `C(xC, yC)`.
Wypisz `Tak`, jeśli punkty **nie są współliniowe** (mogą tworzyć trójkąt), w przeciwnym razie `Nie`.

### Wejście

Sześć liczb całkowitych (każda w osobnej linii):

1. `xA`
2. `yA`
3. `xB`
4. `yB`
5. `xC`
6. `yC`

### Wyjście

Jedno słowo: `Tak` lub `Nie`.

### Przykład

**Wejście:**

```
-3
-2
-3
1
-3
0
```

**Wyjście:**

```
Nie
```

### Uwagi

* Sprawdź pole trójkąta: jeśli równe `0`, punkty są współliniowe.

*/

// Funkcja sprawdzajaca, czy punkty moga stanowic wierzcholki trojkata
function czyTrojkat(punkty) {
  var xA = punkty[0];
  var yA = punkty[1];
  var xB = punkty[2];
  var yB = punkty[3];
  var xC = punkty[4];
  var yC = punkty[5];

  var AB = Math.sqrt(Math.pow(xB - xA, 2) + Math.pow(yB - yA, 2));
  var BC = Math.sqrt(Math.pow(xC - xB, 2) + Math.pow(yC - yB, 2));
  var AC = Math.sqrt(Math.pow(xC - xA, 2) + Math.pow(yC - yA, 2));

  return AB + BC > AC && AB + AC > BC && BC + AC > AB;
}

// Testy

function testCzyTrojkat() {
  console.assert(czyTrojkat([-3, -2, -3, 1, -3, 0]) === false);
  console.assert(czyTrojkat([0, 0, 3, 0, 0, 4]) === true);
  console.assert(czyTrojkat([0, 0, 0, 0, 0, 0]) === false);
  console.assert(czyTrojkat([1, 2, 2, 2, 3, 2]) === false);
  console.assert(czyTrojkat([0, 0, 5, 0, 0, 5]) === true);
}

testCzyTrojkat();

File: src/python/09_listy_wprowadzenie/zad10.py
"""
ZAD-10 — Czy punkty mogą być wierzchołkami trójkąta?

**Poziom:** ★★☆
**Tagi:** `geometria`, `warunki`, `listy`

### Treść

Wczytaj współrzędne trzech punktów `A(xA, yA)`, `B(xB, yB)`, `C(xC, yC)`.
Wypisz `Tak`, jeśli punkty **nie są współliniowe** (mogą tworzyć trójkąt), w przeciwnym razie `Nie`.

### Wejście

Sześć liczb całkowitych (każda w osobnej linii):

1. `xA`
2. `yA`
3. `xB`
4. `yB`
5. `xC`
6. `yC`

### Wyjście

Jedno słowo: `Tak` lub `Nie`.

### Przykład

**Wejście:**

```
-3
-2
-3
1
-3
0
```

**Wyjście:**

```
Nie
```

### Uwagi

* Sprawdź pole trójkąta: jeśli równe `0`, punkty są współliniowe.

"""


def czy_trojkat(lista):
    xA, yA, xB, yB, xC, yC = lista

    a = ((xB - xA) ** 2 + (yB - yA) ** 2) ** 0.5
    b = ((xC - xB) ** 2 + (yC - yB) ** 2) ** 0.5
    c = ((xA - xC) ** 2 + (yA - yC) ** 2) ** 0.5

    if a + b > c and a + c > b and b + c > a:
        return True

    return False


def test_czy_trojkat():
    assert not czy_trojkat([1, 1, 1, 1, 1, 1])
    assert not czy_trojkat([0, 0, 2, -2, 5, -5])
    assert czy_trojkat([-2, 4, 7, 5, 8, -8])


if __name__ == "__main__":

    test_czy_trojkat()
